### DISCUSSION TITLE
Port micropython bluetooth to standard python: one code to rule them all

### DISCUSSION THREAD

**Discussion by vricosti** [micropython/micropython#14260] [General]
Hi,
I am currently building a small device using ble and to prototype I am using a raspberry pi 4B running linux so I wrote some code using python/bluez but now I would like to use a raspbery pico W and I have to rewrite the ble server.
My question is: wouldn't be great to port the micropython/bluetooth to the standard python to write once and run everywhere ?

**Comment by robert-hh** [comment]
Sure. Just go ahead and do it.

**Comment by sosi-deadeye** [comment]
Oh yes, I think it would be very useful addition for other people.

**Comment by jotamudo** [comment]
I sincerely think this will end up as a __bad__ idea, specifically because there are some hard time constraints on the connection maintenance process and having the overhead of the interpreter blocking or slowing it down may present some harsh penalties for the application code using it. I'd guess it could be more approchable if you just made an adaptor of the already done "low-level" ble code
